### PR TITLE
Migrate notifications

### PR DESCRIPTION
**Pull request by achtenhagen**
This PR migrates the `registerDeviceForPushNotifications` function of the Root Service to `registerDeviceForPushNotificationsPayload`, which is now an extension of the Network Manager.

Was able to figure out how the back-end wants the request formatted and updated the code accordingly.

### CODE REVIEW

File: Blockchain/AppDelegate.swift [hunk 1]
```diff
@@ -123,6 +117,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     }
 
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let token = deviceToken.map { String(format: "%02x", $0) }.joined()
```
Comment (bchrisarriola) [inline Blockchain/AppDelegate.swift]: Is this format the same as the one declared in obj-c (which is "%02.2hhX")?

Comment (bchrisarriola) [reply Blockchain/AppDelegate.swift]: Also, maybe move this to an extension of `Data`?

File: Blockchain/Network/Extensions/BlockchainAPI+Payload.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//  Blockchain
+//
+//  Created by Maurice A. on 5/3/18.
+//  Copyright © 2018 Blockchain Luxembourg S.A. All rights reserved.
+//
+
+import Foundation
+
+extension BlockchainAPI {
+    static func registerDeviceForPushNotificationsPayload(_ guid: String, _ sharedKey: String, _ deviceToken: String) -> Data? {
```
Comment (bchrisarriola) [inline Blockchain/Network/Extensions/BlockchainAPI+Payload.swift]: Maybe put this in a struct like `PushNotificationAuthPayload`?

File: Blockchain/RootService.m [hunk 3]
```diff
@@ -561,31 +561,31 @@ - (TabControllerManager *)tabControllerManager
 //    }
 //}
 
-- (void)registerDeviceForPushNotifications
```
Comment (bchrisarriola) [inline Blockchain/RootService.m]: Can also comment out `didRegisterForRemoteNotificationsWithDeviceToken`

File: BlockchainTests/Network/BlockchainAPI+PayloadTests.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+//  BlockchainTests
+//
+//  Created by Maurice A. on 5/4/18.
+//  Copyright © 2018 Blockchain Luxembourg S.A. All rights reserved.
+//
+
+import XCTest
+@testable import Blockchain
+
+class BlockchainAPIPayloadTests: XCTestCase {
```
Comment (bchrisarriola) [inline BlockchainTests/Network/BlockchainAPI+PayloadTests.swift]: 🔥

File: Blockchain/AppDelegate.swift (hunk 1 above)
Comment (achtenhagen) [reply Blockchain/AppDelegate.swift]: Yes same format: see the comments to this answer: https://stackoverflow.com/a/39648407

File: Blockchain/RootService.m (hunk 3 above)
Comment (achtenhagen) [reply Blockchain/RootService.m]: 👍

**Pull request merged**